dask/dataframe/core.py

```diff
@@ -1204,6 +1211,62 @@ def join(self, other, on=None, how='left',
                      left_on=on, suffixes=[lsuffix, rsuffix],
                      npartitions=npartitions)
 
+    @classmethod
+    def _bind_operator_method(cls, name, op):
+        """ bind operator method like DataFrame.add to this class """
+
+        # name must be explicitly passed tp div div method whose name is truediv
+
+        def meth(self, other, axis='columns', level=None, fill_value=None):
+            if level is not None:
+                raise NotImplementedError('level must be None')
+
+            axis = self._validate_axis(axis)
+            left = pd.DataFrame(columns=self.columns)
+
+            right = None
+
+            if axis == 1:
+                # when axis=1, series will be added to each row
+                # it not supported for dd.Series.
+                # dd.DataFrame is not affected as op is applied elemwise
+                if isinstance(other, Series):
+                    msg = 'Unable to {0} dd.Series with axis=1'.format(name)
+                    raise ValueError(msg)
+                elif isinstance(other, pd.Series):
+                    right = pd.DataFrame(columns=other.index)
+            if isinstance(other, (DataFrame, pd.DataFrame)):
+                right = pd.DataFrame(columns=other.columns)
+
+            if right is not None:
+                columns = op(left, right, axis=axis).columns.tolist()
+            else:
+                columns = self.columns
+
+            return map_partitions(op, columns, self, other,
+                                  axis=axis, fill_value=fill_value)
+        meth.__doc__ = op.__doc__
+        bind_method(cls, name, meth)
+
+# bind operators
+for op in [operator.abs, operator.add, operator.and_, operator_div,
+           operator.eq, operator.gt, operator.ge, operator.inv,
+           operator.lt, operator.le, operator.mod, operator.mul,
+           operator.ne, operator.neg, operator.or_, operator.pow,
+           operator.sub, operator.truediv, operator.floordiv, operator.xor]:
+    _Frame._bind_operator(op)
+    Scalar._bind_operator(op)
+
+for name in ['add', 'sub', 'mul', 'div',
+             'truediv', 'floordiv', 'mod', 'pow',
+             'radd', 'rsub', 'rmul', 'rdiv',
+             'rtruediv', 'rfloordiv', 'rmod', 'rpow']:
+    meth = getattr(pd.DataFrame, name)
+    DataFrame._bind_operator_method(name, meth)
+
+    meth = getattr(pd.Series, name)
+    Series._bind_operator_method(name, meth)
```

Comment:
With regards to [this comment](https://github.com/blaze/dask/pull/625#issuecomment-134993136) I think that we should keep dataframe operators the way that you've built them.  Your approach fits `dask.dataframe` really well.  This is a more complex case than `dask.array`.